### ISSUE TITLE
GameContent.StoryStructuresSpawnConditions crash

### ISSUE THREAD

**Issue by DejFidOFF** [anegostudios/VintageStory-Issues#4678]
### Game Version

v1.20tc1

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

When try to write to book it disconnected me from server

### How to reproduce

get book & ink and quil try to write to book.

### Logs
```
[Error] Exception: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.StoryStructuresSpawnConditions.GetStoryStructureAt(BlockPos pos) in VSSurvivalMod\Lore\StoryStructuresSpawnConditions.cs:line 132
   at Vintagestory.GameContent.StoryStructureProtection.Event_OnTestBlockAccess(IPlayer player, BlockSelection blockSel, EnumBlockAccessFlags accessType, String& claimant, EnumWorldAccessResponse response) in VSSurvivalMod\Lore\StoryStructureProtection.cs:line 51
   at Vintagestory.Server.ServerSystemBlockSimulation.HandleBlockInteract(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\World\BlockSimulation.cs:line 429
   at Vintagestory.Server.ServerSystemInventory.HandleHandInteraction(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\Inventory.cs:line 255
   at Vintagestory.Server.ServerMain.ProcessNetMessage(NetIncomingMessage msg, NetServer mainSocket) in VintagestoryLib\Server\ServerMainNetworking.cs:line 141
```

**Comment by Craluminum2413** [comment]
It is trying to generate story location, thus it is not caused by book

**Comment by DejFidOFF** [comment]
> The game is trying to generate story location, thus it is not caused by book

Odd. Its fresh map just made and no one play it.

**Comment by DejFidOFF** [comment]
> The game is trying to generate story location, thus it is not caused by book

reading signed book causing to generate story location. Every player which join server and read signed book guide - client crashed and generate Village location and make a mark on the map.

**Comment by tyronx** [comment]
thanks for the report but this should be fixed in rc2. Please reopen if its still crashing on the latest version

**Comment by DejFidOFF** [comment]
> thanks for the report but this should be fixed in rc2. Please reopen if its still crashing on the latest version

Server is running VS 1.20rc5 ( every version I deleted all files and make fresh install ) and still first read of signed guidbook causing player kick from the server and causing client crash... After he rejoin can read book without any issues.

18.12.2024 18:16:38 [Event] Player Abby_Pyrrho got removed. Reason: Threw an exception at the server
18.12.2024 18:16:38 [Error] Exception: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.StoryStructuresSpawnConditions.GetStoryStructureAt(BlockPos pos) in VSSurvivalMod\Lore\StoryStructuresSpawnConditions.cs:line 132
   at Vintagestory.GameContent.StoryStructureProtection.Event_OnTestBlockAccess(IPlayer player, BlockSelection blockSel, EnumBlockAccessFlags accessType, String& claimant, EnumWorldAccessResponse response) in VSSurvivalMod\Lore\StoryStructureProtection.cs:line 60
   at Vintagestory.Server.ServerSystemBlockSimulation.HandleBlockInteract(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\World\BlockSimulation.cs:line 429
   at Vintagestory.Server.ServerSystemInventory.HandleHandInteraction(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\Inventory.cs:line 255
   at Vintagestory.Server.ServerMain.ProcessNetMessage(NetIncomingMessage msg, NetServer mainSocket) in VintagestoryLib\Server\ServerMainNetworking.cs:line 141

**Comment by tyronx** [comment]
oh interesting. Is this a world created in any of the pre-releases?

**Comment by DejFidOFF** [comment]
> oh interesting. Is this a world created in any of the pre-releases?

unfortunately yes, I made this map in 1.20rc1 :(

**Comment by DejFidOFF** [comment]
This error also causing reading any lore scroll

**Comment by inshyian** [comment]
I have the error when trying to open the book which I got through panning; my client disconnects from the server, and on the server side there is an error: "Exception: Object reference not set to an instance of an object. ..." Actually the server does not crashes and I can reconnect to the server, and if I try to open the book again it opens successfully.
The error happens not every time when I open the book. Part of books successfully opened after I got it via panning, other part opened after reconnect.

How to reproduce:
1. Got book by panning
2. Right click on the book
3. Displays header of the story
4. I can check the new entry in journal, and it always shows as intended, I can read it
5. When I do right click on the book again then it can disconnect me from the server; but sometimes book opens and I can read it
6. If I was disconnected on previously step, and when I reconnect I can open the book and read it

Another way to recreate:
1. Previously found and opened book stored in bookshelf
2. Took one
3. Try to open, the same error can be happen here but sometimes book opens normally (same as in step 5)
4. When I try to reconnect to the server and open it again looks like it always opens (mb it can happen after reconnect too, I did not tested).

Game version is 1.20rc6.
My world was created in 1.20rc1 on my client, then moved to the dedicated server in local network. Books found in rc6.

The error I have in my server log, it happened 7 time. Seems it a bit differs from the error posted by @DejFidOFF:
```
Dec 25 02:11:14 deb13100 VSSRV[182960]: 25.12.2024 02:11:14 [Server Error] Exception: Object reference not set to an instance of an object.
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.GameContent.StoryStructuresSpawnConditions.GetStoryStructureAt(BlockPos pos) in VSSurvivalMod\Lore\StoryStructuresSpawnConditions.cs:line 132
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.GameContent.StoryStructureProtection.Event_OnTestBlockAccess(IPlayer player, BlockSelection blockSel, EnumBlockAccessFlags accessType, String& claimant, EnumWorldAccessResponse response) in VSSurvivalMod\Lore\StoryStructureProtection.cs:line 60
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.Server.ServerSystemBlockSimulation.HandleBlockInteract(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\World\BlockSimulation.cs:line 429
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.Server.ServerSystemInventory.HandleHandInteraction(Packet_Client packet, ConnectedClient client) in VintagestoryLib\Server\Systems\Inventory.cs:line 255
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.Server.ServerMain.HandleClientPacket(ConnectedClient client, Byte[] data) in VintagestoryLib\Server\ServerMainNetworking.cs:line 154
Dec 25 02:11:14 deb13100 VSSRV[182960]:    at Vintagestory.Server.ServerMain.ProcessNetMessage(NetIncomingMessage msg, NetServer mainSocket) in VintagestoryLib\Server\ServerMainNetworking.cs:line 147
```
I can't reproduce the error now, but yesterday evening it happen many times. If it happens again I'll add a screenshot of the text I had on the client side.

**Comment by DejFidOFF** [comment]
Unfortunately it kicks players from the server.

**Comment by radfast** [comment]
Thanks for the reports, very helpful. This is, with a bit of luck, fixed from next build upcoming (1.20.0-rc.7)
